internal/rbd/rbd_attach.go

```diff
@@ -388,6 +388,33 @@ func appendKRbdDeviceTypeAndOptions(cmdArgs []string, userOptions string) []stri
 	return cmdArgs
 }
 
+// appendCrushLocationMapOptions converts crushLocationMap to the format
+// "read_from_replica=localize,crush_location=key1:value1|key2:value2"
+// and appends it to mapOptions.
+func appendCrushLocationMapOptions(mapOptions string, crushLocationMap map[string]string) string {
+	if len(crushLocationMap) == 0 {
+		// if no crush locations are found, return the mapOptions as it is.
+		return mapOptions
+	}
+
+	var b strings.Builder
+	b.WriteString("read_from_replica=localize,crush_location=")
+	first := true
+	for key, val := range crushLocationMap {
```

Comment:
`key` and `val` may not contain `,`, else things will go wrong. This is use-input, so you'll need to check it more carefully. There might be other characters that are forbidden.

Reply:
> `key` and `val` may not contain `,`, else things will go wrong. This is use-input, so you'll need to check it more carefully. There might be other characters that are forbidden.

@idryomov 
Any suggestion on validating this ? 
It'll be good to know what characters are valid and what's not in key:val parts

Reply:
kubernetes already has a check in place, so we should be good.

```
$ k label node minikube hello.io/he="abcd ede"
error: invalid label value: "hello.io/he=abcd ede": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')

$ k label node minikube "hello.io/hn dsfds"="abcdede"
The Node "minikube" is invalid: metadata.labels: Invalid value: "hello.io/hn dsfds": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
```